flac/decode.go

```diff
@@ -23,7 +23,14 @@ func Decode(r io.Reader) (s beep.StreamSeekCloser, format beep.Format, err error
 			}
 		}
 	}()
-	d.stream, err = flac.New(r)
+
+	rs, seeker := r.(io.ReadSeeker)
```

Comment:
I think this part needs some thought.  

Calling flac.Seek results in the use of a bufio.Buffer and NewSeek does not.  According to the kindly mewkiz/flac folks (@mewmew) this could result in performance issues if *os.File is passed in as the io.Reader.  It might be better to enable Seek more explicitly but I'm not sure how that might be implemented cleanly.  Thoughts?

Reply:
It's probably fine for now. Calling seek won't work on a bufio.Reader as you mentioned. Lets first see if anyone run into real world performance issues with this, and then we can try to think of a solution.

Great to seek seek support added in Beep! Good job @cswank.

Reply:
That sounds good to me.  Thanks @mewmew .